windows: fsevents v1 is getting built

When doing an install of dependecies on my windows based system, fsevents 1.2.11 attempts to run node-gyp, asking for Python.  When Python is not found  node-gyp sets the exitcode to npm as non-zero. This causes our CI to fail the build.

It seems that fsevents could do a check ahead of time for a darwin based system before calling node-gyp since fsevents is strictly for darwin systems.

Below is the captured output:
```
> fsevents@1.2.11 install C:\Users\jchappell\Projects\wida\alkami.client.widget.myaccountsv3\node_modules\fsevents
> node-gyp rebuild


C:\Users\jchappell\Projects\wida\alkami.client.widget.myaccountsv3\node_modules\fsevents>if not defined npm_config_node_gyp (node "C:\Program Files\nodejs\node_modules\npm\node_modules\npm-lifecycle\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node "C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\bin\node-gyp.js" rebuild )
gyp info it worked if it ends with ok
gyp info using node-gyp@3.8.0
gyp info using node@10.16.3 | win32 | x64
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Program Files\nodejs\node_modules\npm\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at C:\Program Files\nodejs\node_modules\npm\node_modules\graceful-fs\polyfills.js:282:31
gyp ERR! stack     at FSReqWrap.oncomplete (fs.js:153:21)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Program Files\\nodejs\\node_modules\\npm\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild"
gyp ERR! cwd C:\Users\jchappell\Projects\wida\alkami.client.widget.myaccountsv3\node_modules\fsevents
gyp ERR! node -v v10.16.3
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
```

is this problem just apper into windows CI enviroments? now i am try to build in a CI/CD Linux eviroments and give some similar errors

i am using node 10.5.1, npm 6.0.1 and fsevent 1.2.11 (i was using fsevents 1.2.9 but the same error is used)

`ui-dev/node_modules/.staging
npm timing stage:rollbackFailedOptional Completed in 412ms
npm timing stage:runTopLevelLifecycles Completed in 76478ms
npm verb stack Error: Unsupported platform for fsevents@1.2.11: wanted {"name":"fsevents","version":"1.2.11","description":"Native Access to Mac OS-X FSEvents","main":"fsevents.js","dependencies":{"bindings":"^1.5.0","nan":"^2.12.1","node-pre-gyp":"*"},"os":["darwin"],"engines":{"node":">=4.0"},"scripts":{"test":"node ./test/fsevents.js && node ./test/function.js 2> /dev/null","install":"node-gyp rebuild"},"repository":{"type":"git","url":"git+https://github.com/strongloop/fsevents.git"},"keywords":["fsevents","mac"],"author":{"name":"Philipp Dunkel","email":"pip@pipobscure.com"},"license":"MIT","bugs":{"url":"https://github.com/strongloop/fsevents/issues"},"homepage":"https://github.com/strongloop/fsevents","_resolved":"https://npm-registry.fif.tech/fsevents/-/fsevents-1.2.11.tgz","_integrity":"sha512-+ux3lx6peh0BpvY0JebGyZoiR4D+oYzdPZMKJwkZ+sFkNJzpL7tXc/wehS49gUAxg3tmMHPHZkA8JU2rhhgDHw==","_from":"fsevents@1.2.11","gypfile":true,"bundleDependencies":["node-pre-gyp"],"_id":"fsevents@1.2.11","_requested":{"type":"version","registry":true,"raw":"fsevents@1.2.11","name":"fsevents","escapedName":"fsevents","rawSpec":"1.2.11","saveSpec":null,"fetchSpec":"1.2.11"},"_spec":"1.2.11","_where":"/var/tmp/workspace/cmr/cl/apertura/web-ui-dev","_args":[["fsevents@1.2.11","/var/tmp/workspace/cmr/cl/apertura/web-ui-dev"]]} (current: {"os":"linux","cpu":"x64"})
npm verb stack     at checkPlatform (/usr/local/lib/node_modules/npm/node_modules/npm-install-checks/index.js:45:14)
npm verb stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
npm verb stack     at ret (eval at makeNodePromisifiedEval (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promisify.js:184:12), <anonymous>:13:39)
npm verb stack     at readJson.then.catch.then (/usr/local/lib/node_modules/npm/lib/install/action/refresh-package-json.js:31:12)
npm verb stack     at tryCatcher (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/util.js:16:23)
npm verb stack     at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:512:31)
npm verb stack     at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:569:18)
npm verb stack     at Promise._settlePromise0 (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
npm verb stack     at Promise._settlePromises (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:693:18)
npm verb stack     at Promise._fulfill (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:638:18)
npm verb stack     at Promise._settlePromise (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:582:21)
npm verb stack     at Promise._settlePromise0 (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:614:10)
npm verb stack     at Promise._settlePromises (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:693:18)
npm verb stack     at Promise._fulfill (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:638:18)
npm verb stack     at Promise._resolveCallback (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:432:57)
npm verb stack     at Promise._settlePromiseFromHandler (/usr/local/lib/node_modules/npm/node_modules/bluebird/js/release/promise.js:524:17)
npm verb pkgid fsevents@1.2.11
npm verb cwd /var/tmp/workspace/cmr/cl/apertura/web-ui-dev
npm verb Linux 3.10.0-1062.12.1.el7.x86_64
npm verb argv "/usr/local/bin/node" "/usr/local/bin/npm" "install" "--verbose" "--userconfig=.npmrc"
npm verb node v10.5.0
npm verb npm  v6.1.0
npm ERR! code EBADPLATFORM
npm ERR! notsup Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"linux","arch":"x64"})
npm ERR! notsup Valid OS:    darwin
npm ERR! notsup Valid Arch:  any
npm ERR! notsup Actual OS:   linux
npm ERR! notsup Actual Arch: x64
npm verb exit [ 1, true ]
npm timing npm Completed in 80960ms`

i check dependencies with npm list fsevent and another package use lowest version of fsevent like chokidar, any advice?

--- a/install.js
+++ b/install.js
@@ -1,0 +1,13 @@
+const { spawn } = require('child_process');
+
+const rebuildIfDarwin = () => {
+  if (process.platform !== 'darwin') {
+    console.log();
+    console.log(`Skipping 'fsevents' build as platform ${process.platform} is not supported`);
+    process.exit(0);
+  } else {
+    spawn('node-gyp', ['rebuild'], { stdio: 'inherit' });
+  }
+};
+
+rebuildIfDarwin();

--- a/package.json
+++ b/package.json
@@ -15,7 +15,8 @@
     "node": ">= 4.0"
   },
   "scripts": {
-    "test": "node ./test/fsevents.js && node ./test/function.js 2> /dev/null"
+    "test": "node ./test/fsevents.js && node ./test/function.js 2> /dev/null",
+    "install": "node install.js"
   },
   "repository": {
     "type": "git",

